oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreBuilder.java

```diff
@@ -534,13 +534,13 @@ public DiffCache getDiffCache(int clusterId) {
      * @return this
      */
     public T setBlobStore(BlobStore blobStore) {
-        this.blobStoreSupplier = ofInstance(blobStore);
+        this.blobStoreSupplier = () -> blobStore;
         return thisBuilder();
     }
 
     public BlobStore getBlobStore() {
         if (blobStoreSupplier == null) {
-            blobStoreSupplier = ofInstance(new MemoryBlobStore());
+            blobStoreSupplier = () -> new MemoryBlobStore();
```

Comment:
It should be replaced with `memoize` as well.

Reply:
Why? It seems that although this looks similar, the actual requirements are different.

Reply:
ack :-)